src/MeshVolume.js

```diff
@@ -171,6 +182,61 @@ export default class MeshVolume {
     return this.meshrep[channel].userData.isovalue;
   }
 
+  updateClipRegion(xmin, xmax, ymin, ymax, zmin, zmax) {
+    // incoming values expected to be between 0 and 1.
+    // I shift them here to be between -0.5 and 0.5
+    this.bounds = {
+      bmin: new Vector3(xmin - 0.5, ymin - 0.5, zmin - 0.5),
+      bmax: new Vector3(xmax - 0.5, ymax - 0.5, zmax - 0.5),
+    };
+    this.updateClipFromBounds();
+  }
+
+  updateClipFromBounds() {
+    const xmin = this.bounds.bmin.x;
+    const ymin = this.bounds.bmin.y;
+    const zmin = this.bounds.bmin.z;
+    const xmax = this.bounds.bmax.x;
+    const ymax = this.bounds.bmax.y;
+    const zmax = this.bounds.bmax.z;
+
+    const euler = this.meshPivot.rotation;
+
+    for (let channel = 0; channel < this.meshrep.length; ++channel) {
+      if (!this.meshrep[channel]) {
+        continue;
+      }
+      const planes = [];
+      // up to 6 planes.
+      if (xmin > -0.5) {
+        planes.push(new Plane(new Vector3(1, 0, 0).applyEuler(euler), this.meshRoot.position.x + -xmin * this.scale.x));
+      }
+      if (ymin > -0.5) {
+        planes.push(new Plane(new Vector3(0, 1, 0).applyEuler(euler), this.meshRoot.position.y + -ymin * this.scale.y));
+      }
+      if (zmin > -0.5) {
+        planes.push(new Plane(new Vector3(0, 0, 1).applyEuler(euler), this.meshRoot.position.z + -zmin * this.scale.z));
+      }
+      if (xmax < 0.5) {
+        planes.push(new Plane(new Vector3(-1, 0, 0).applyEuler(euler), this.meshRoot.position.x + xmax * this.scale.x));
+      }
+      if (ymax < 0.5) {
+        planes.push(new Plane(new Vector3(0, -1, 0).applyEuler(euler), this.meshRoot.position.y + ymax * this.scale.y));
+      }
+      if (zmax < 0.5) {
+        planes.push(new Plane(new Vector3(0, 0, -1).applyEuler(euler), this.meshRoot.position.z + zmax * this.scale.z));
+      }
```

Comment:
Maybe I'm missing something..... but can lines 209-228 be pulled outside of the for loop? Doesn't look like they depend on `channel`?

Reply:
They don't depend on channel but they need to be added once for each channel that has a mesh representation (because each single mesh needs to have its own clip planes set in threejs)

Reply:
I guess I could precompute them outside the loop and then add them inside the loop, of course!

Reply:
Yes, that's what I meant, I wasn't sure how many channels we might have or how much this would matter. Your call!